pghoard/basebackup.py

```diff
@@ -756,13 +824,46 @@ def run_local_tar_basebackup(self, delta=False):
                         pgdata=pgdata, tablespaces=tablespaces, target_chunk_size=target_chunk_size
                     )
                     chunks_count = len(chunks)
+
+                    delta_stats: Optional[Dict[str, int]] = None
+                    if with_delta_stats:
+                        delta_stats = {}
+
                     # Tar up the chunks and submit them for upload; note that we start from chunk 1 here; chunk 0
                     # is reserved for special files and metadata and will be generated last.
-                    chunk_files = self.create_and_upload_chunks(chunks, data_file_format, temp_base_dir)
+                    chunk_files = self.create_and_upload_chunks(
+                        chunks, data_file_format, temp_base_dir, delta_stats=delta_stats
+                    )
 
                     total_size_plain = sum(item["input_size"] for item in chunk_files)
                     total_size_enc = sum(item["result_size"] for item in chunk_files)
 
+                    if with_delta_stats:
+                        existing_hashes = self.fetch_all_data_files_hashes()
+                        new_hashes = {k: delta_stats[k] for k in set(delta_stats).difference(set(existing_hashes))}
+                        control_files_metadata_extra["delta_stats"] = {"new_hashes": new_hashes}
+
+                        planned_upload_size = sum(new_hashes.values())
+                        planned_upload_count = len(new_hashes)
+                        planned_total_size = sum(existing_hashes.values())
+                        planned_total_count = len(existing_hashes)
+
+                        if existing_hashes:
+                            self.metrics.gauge(
+                                "pghoard.planned_delta_backup_changed_data_files_ratio",
+                                planned_upload_count / planned_total_count
+                            )
+                            if planned_total_size:
+                                self.metrics.gauge(
+                                    "pghoard.planned_delta_backup_changed_data_size_ratio",
+                                    planned_upload_size / planned_total_size
+                                )
+
+                        self.metrics.increase("pghoard.planned_delta_backup_total_size", inc_value=planned_upload_size)
+                        self.metrics.gauge("pghoard.planned_delta_backup_upload_size", planned_upload_size)
+                        self.metrics.increase("pghoard.planned_delta_backup_total_files", inc_value=planned_upload_count)
```

Comment:
there is a gauge already (next lime), here the idea was to keep the total amount of uploaded file as well (it will accumulate amount of all files ever uploaded), so we can easily calculate the total money spent on storage API calls for example.

Reply:
Ah, right